src/formats/SDF.cpp

```diff
@@ -361,21 +361,20 @@ optional<uint64_t> SDFFormat::forward() {
         nbonds = parse<size_t>(counts_line.substr(3,3));
     } catch (const Error&) {
         // We could not read an integer, so give up here
-        throw format_error("could not parse counts line: '{}'", counts_line);
+        throw format_error("could not parse counts line in SDF file: '{}'", counts_line);
     }
 
-    try {
-        for (size_t i=0; i<(natoms + nbonds); i++) {
-            file_.readline();
-        }
-    } catch (const FileError&) {
-        // We could not read the lines from the file
-        throw format_error(
+    for (size_t i=0; i<(natoms + nbonds); i++) {
+        if (file_.eof()) {
+            throw format_error(
             "not enough lines in '{}' for SDF format", file_.path()
         );
+        }
+        file_.readline();
     }
 
-    // Search for ending character, updating the position in the file only
+    // Search for ending character, updating the cursor in the file for the next
+    // call to forward
     while (!file_.eof()) {
         if (file_.readline() == "$$$$") {
             break;
```

Comment:
right, I only looked at the forward methods, but it seems that this is also needed in other places. Good catch!!

Reply:
Actually, we should not need to check for EOF here, since forward already does check for us. The only possible issue would be that the file changed on disk between the call to forward and the call to `read_next`.